### PR TITLE
Remove bluemonday

### PR DESCRIPTION
**Pull request by david-bezero**
This was previously used:

- (wrongly) for some JSON escaping (fixed by replacing with real JSON serialisation), and
- for building an email template (which also had a vulnerability because it didn't apply to the `site` parameter).

Both have been fixed properly here. Fortunately we're not using the email capability at all. Having seen this code makes me super uncomfortable about trusting these authors with anything authentication-related though…

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,8 +11,8 @@ require (
 	github.com/go-pkgz/rest v1.16.0
 	github.com/golang-jwt/jwt v3.2.2+incompatible
 	github.com/golang-jwt/jwt/v4 v4.4.1
-	github.com/microcosm-cc/bluemonday v1.0.20
 	github.com/nullrocks/identicon v0.0.0-20180626043057-7875f45b0022
+	github.com/pkg/errors v0.9.1
```
Comment (david-bezero) [inline go.mod:15]: this appeared when running `go mod tidy` - seems some of the oauth changes brought it in

File: provider/custom_server.go [hunk 2]
```diff
@@ -162,16 +162,19 @@ func (c *CustomServer) handleUserInfo(w http.ResponseWriter, r *http.Request) {
...
-	ava := p.Sanitize(fmt.Sprintf(c.URL+"/avatar?user=%s", userID))
-	res := fmt.Sprintf(`{
-					"id": "%s",
-					"name":"%s",
-					"picture":"%s"
-					}`, userID, userID, ava)
+	user := token.User{
+		ID:      userID,
+		Name:    userID,
+		Picture: fmt.Sprintf(c.URL+"/avatar?user=%s", userID),
```
Comment (david-bezero) [inline provider/custom_server.go:168]: vulnerability fixed: backslashes and double quotes in `userID` were not being escaped here

File: provider/verify.go [hunk 3]
```diff
@@ -179,10 +176,10 @@ func (e VerifyHandler) sendConfirmation(w http.ResponseWriter, r *http.Request)
...
 		Site    string
 	}{
-		User:    user,
-		Address: address,
-		Token:   tkn,
-		Site:    r.URL.Query().Get("site"),
+		User:    safeHTML(user),
+		Address: safeHTML(address),
+		Token:   template.HTMLEscapeString(tkn),
+		Site:    template.HTMLEscapeString(site),
```
Comment (david-bezero) [inline provider/verify.go:182]: vulnerability fixed: `site` comes from a URL parameter and was not being HTML escaped at all. Token is less of an issue, but I added HTML escaping just-in-case. The user and address also had some "prettifying" and truncation logic (presumably for plain-text viewing), which I have preserved in `safeHTML`, though I don't really see the point.

File: provider/verify.go [hunk 4]
```diff
@@ -150,7 +147,7 @@ func (e VerifyHandler) sendConfirmation(w http.ResponseWriter, r *http.Request)
 		},
 		SessionOnly: r.URL.Query().Get("session") != "" && r.URL.Query().Get("session") != "0",
 		StandardClaims: jwt.StandardClaims{
-			Audience:  e.sanitize(r.URL.Query().Get("site")),
+			Audience:  site,
```
Comment (david-bezero) [inline provider/verify.go:150]: this was incorrectly being HTML encoded, but never appears in any HTML document

**Review by alek-sys — APPROVED** [review]

**Pull request merged**